src/gates_function.jl

```diff
@@ -0,0 +1,60 @@
+using ITensors
+include("constants.jl")
+
+# Expected units of the quantities used in gates                                                                    
+# del_m2 = ergs squared
+# hbar =  erg s
+# c = cm/s
+# G_F = erg cm^3           
+# ∇x = cm
+# E = erg
+
+function create_gates(s, n, ω, B, N, ∇x, τ)
+    # Make gates (1,2),(2,3),(3,4),... i.e. unitary gates which act on any (non-neighboring) pairs of sites in the chain.
+    # Create an empty ITensors array that will be our Trotter gates
+    gates = ITensor[]                                                              
+    
+    for i in 1:(N-1)
+        for j in i+1:N
+            #s1, s2 are non-neighbouring spin site/indices from the s array
+            s1 = s[i]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        
+            s2 = s[j]
+            # assert B vector to have a magnitude of 1 while preserving its direction.
+            @assert norm(B) == 1
+
+            # Our neutrino system Hamiltonian of self-interaction term represents 1D Heisenberg model.
+            # total Hamiltonian of the system is a sum of local terms hj, where hj acts on sites i and j which are paired for gates to latch onto.
+            # op function returns these operators as ITensors and we tensor product and add them together to compute the operator hj.
+            # ni and nj are the neutrions at site i and j respectively.
+            # mu pairs divided by 2 to avoid double counting
+            
+            hj = 
+            ((2.0* √2 * G_F * (n[i]+ n[j])/(2*((∇x)^3)) * 1/N) * 
+            (op("Sz", s1) * op("Sz", s2) +
+             1/2 * op("S+", s1) * op("S-", s2) +
+             1/2 * op("S-", s1) * op("S+", s2)))
+             
+             if ω[i] != 0
+                hj += (1/(N-1))* (ω[i] *
```

Comment:
I think you need to be careful to apply w[i] to only site i, and w[j] to only site j. Currently everything works because all of your w's have the same value. But I think this should go inside the parentheses and multiply each term separately. Does that make sense?